### DISCUSSION TITLE
DnsCrypt configuration on Ubuntu 20.04 lts

### DISCUSSION THREAD

**Discussion by Leonis87** [DNSCrypt/dnscrypt-proxy#1854] [Q&A]
Hi,
I'm new in Dnscrypt, can anybody give me good advise how to configure DnsCrypt on Ubuntu 20.04 lts. I know how to install it and basic configuration, but acording github wiki installation instructions I have to edit and modify resolv.conf file, NetworkMnager.conf file. I would like ask whether is it really necessary ?

Also when I want to use Mullvad VPN, do I have to disable DnsCrypt ?
I would appreciate any advise in terms of simplicity and functionality.
Thanks

**Comment by gnu-like** [comment]
After installing `sudo apt install dnscrypt-proxy` you should disable & remove dnscrypt-proxy.socket :
do these: 
```
$ sudo systemctl stop dnscrypt-proxy.service
$ sudo systemctl stop dnscrypt-proxy.socket
$ sudo systemctl disable dnscrypt-proxy.socket
$ sudo systemctl disable dnscrypt-proxy.service
$ sudo rm /lib/systemd/system/dnscrypt-proxy.service /lib/systemd/system/dnscrypt-proxy.socket
$ sudo rm /var/lib/systemd/deb-systemd-helper-enabled/dnscrypt-proxy.service.dsh-also /var/lib/systemd/deb-systemd-helper-enabled/dnscrypt-proxy.socket.dsh-also /var/lib/systemd/deb-systemd-helper-enabled/multi-user.target.wants/dnscrypt-proxy.service /var/lib/systemd/deb-systemd-helper-enabled/sockets.target.wants/dnscrypt-proxy.socket
$ sudo systemctl daemon-reload
$ sudo systemctl reset-failed
$ sudo dnscrypt-proxy -config /etc/dnscrypt-proxy/dnscrypt-proxy.toml -service install
$ sudo service dnscrypt-proxy start
```
Edit `sudo vim /etc/dnscrypt-proxy/dnscrypt-proxy.toml`
listen_addresses = [] **change it to** listen_addresses = ['127.0.2.1:53']

please pay attention: 
/etc/resolv.conf is symlinked to /run/resolvconf/resolv.conf so remove symlink (`sudo rm /etc/resolv.conf`) and write you own version of /etc/resolv.conf
 `sudo vim /etc/resolv.conf `add nameserver 127.0.2.1 
restart dnscrypt-proxy service `sudo systemctl restart dnscrypt-proxy` now it should work on ubuntu.

**Reply by Leonis87** [reply]
Thanks.
Ok I managed install it manually from binaries. From what I digged out on google I understood that this way I don't have to remove dnscrypt socket. 
But now I'm not sure about "ss -lp 'sport = :domain'" and "systemctl disable systemd-resolved". Do I have to do that ? Because everytime I disable system-resolved I have no internet connection until I restart computer. After restart dnscrypt works alright. 

Also do I have to edit dnsmasq file ? Because on ubuntu 20.04 it is located in different directory than on github wiki installation page. And if I install dnscrypt-proxy from binaries it just dont create any dnsmasq file in etc directories.

**Reply by gnu-like** [reply]
Nope ! there is no need "systemctl disable systemd-resolved" you need remove only "dnscrypt-proxy.socket" and you don't need "dnsmasq" as I said before you just add dns crypt (nameserver 127.0.2.1) or google dns (nameserver 8.8.8.8) manually to /etc/resolv.conf by removing symlinked  /run/resolvconf/resolv.conf from /etc/resolv.conf
